src/bar.cpp

```diff
@@ -505,15 +522,44 @@ waybar::Bar::Bar(struct waybar_output* w_output, const Json::Value& w_config)
     surface_impl_ = std::make_unique<RawSurfaceImpl>(window, *output);
   }
 
-  surface_impl_->setLayer(layer_);
-  surface_impl_->setExclusiveZone(exclusive);
   surface_impl_->setMargins(margins_);
-  surface_impl_->setPassThrough(passthrough);
   surface_impl_->setPosition(position);
   surface_impl_->setSize(width, height);
 
+  /* Init "default" mode from globals */
+  auto& default_mode = configured_modes[MODE_DEFAULT];
+  if (config["layer"] == "top") {
+    default_mode.layer = bar_layer::TOP;
+  } else if (config["layer"] == "overlay") {
+    default_mode.layer = bar_layer::OVERLAY;
+  }
+
+  if (config["exclusive"].isBool()) {
+    default_mode.exclusive = config["exclusive"].asBool();
+  }
+
+  if (config["passthrough"].isBool()) {
+    default_mode.passthrough = config["passthrough"].asBool();
```

Comment:
passthrough is not documented yet, right?

Reply:
It's documented, both in the man and in the wiki. Although I think it shouldn't be now that we have modes support.